{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "governing-indian",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import tensorflow as tf\n",
    "from keras.models import Sequential\n",
    "from keras.layers import Dense\n",
    "from keras.wrappers.scikit_learn import KerasRegressor\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.metrics import r2_score\n",
    "import sklearn\n",
    "from sklearn import preprocessing\n",
    "from tensorflow.keras.optimizers import Adam\n",
    "from sklearn.model_selection import StratifiedKFold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "female-jumping",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_ew2=pd.read_csv('KiKNet_smoothFAS_updated_28Dec20_EW2.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "greenhouse-subsection",
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('display.max_columns', None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "stylish-mechanics",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Unnamed: 0', 'Unnamed: 0.1', 'X', 'Code', 'Address', 'Vs30', 'Z1', 'MT_Magnitude_', 'MT_Depth_', 'rrup_0', 'rrup_1', 'rjb_0', 'rjb_1', 'rhypo_0', 'rhypo_1', 'repi_0', 'repi_1', 'depth', 'Origin_day_', 'originDate', 'Origin_Time_', 'Longitude_', 'Latitude_', 'Region_Name_', 'samplingFreq', 'lon', 'lat', 'Tectonic_Garcia_', 'Tectonic_Zhoa_', 'JMA_Depth_', 'Focal_mechanism_kegan_', 'Strike_', 'recordDate', 'station Lon', 'station Lat', 'origin Time', 'Number_of_Stations_', 'Mo_', 'scale Factor', 'Dip_', 'direction', 'Ztor_0', 'Ztor_1', 'Focal_mechanism_Garcia_', 'Rake_', 'duration Time', 'mag', 'snr_EW1', 'snr_NS1', 'snr_EW2', 'snr_NS2', 'fLow_EW1', 'fLow_NS1', 'fLow_EW2', 'fLow_NS2', 'fHigh_EW1', 'fHigh_NS1', 'fHigh_EW2', 'fHigh_NS2', 'pga_EW1', 'pga_NS1', 'pga_EW2', 'pga_NS2', 'Ia_EW1', 'Ia_NS1', 'Ia_EW2', 'Ia_NS2', 'D5_75_EW1', 'D5_75_NS1', 'D5_75_EW2', 'D5_75_NS2', 'D5_75_sorted_EW1', 'D5_75_sorted_NS1', 'D5_75_sorted_EW2', 'D5_75_sorted_NS2', 'D20_80_EW1', 'D20_80_NS1', 'D20_80_EW2', 'D20_80_NS2', 'D5_95_EW1', 'D5_95_NS1', 'D5_95_EW2', 'D5_95_NS2', 'Vs5', 'Vs10', 'Vs20', 'Vs800', 'Vs50', 'Vs100', 'station', 'f_0.1', 'f_0.106', 'f_0.112', 'f_0.118', 'f_0.125', 'f_0.132', 'f_0.14', 'f_0.148', 'f_0.156', 'f_0.165', 'f_0.175', 'f_0.185', 'f_0.195', 'f_0.206', 'f_0.218', 'f_0.231', 'f_0.244', 'f_0.258', 'f_0.273', 'f_0.289', 'f_0.305', 'f_0.323', 'f_0.341', 'f_0.361', 'f_0.381', 'f_0.403', 'f_0.426', 'f_0.451', 'f_0.477', 'f_0.504', 'f_0.533', 'f_0.563', 'f_0.596', 'f_0.63', 'f_0.666', 'f_0.704', 'f_0.745', 'f_0.787', 'f_0.833', 'f_0.88', 'f_0.931', 'f_0.984', 'f_1.041', 'f_1.1', 'f_1.163', 'f_1.23', 'f_1.301', 'f_1.375', 'f_1.454', 'f_1.538', 'f_1.626', 'f_1.719', 'f_1.818', 'f_1.922', 'f_2.032', 'f_2.149', 'f_2.272', 'f_2.402', 'f_2.54', 'f_2.686', 'f_2.84', 'f_3.003', 'f_3.175', 'f_3.357', 'f_3.55', 'f_3.753', 'f_3.969', 'f_4.196', 'f_4.437', 'f_4.691', 'f_4.96', 'f_5.245', 'f_5.546', 'f_5.864', 'f_6.2', 'f_6.556', 'f_6.932', 'f_7.329', 'f_7.75', 'f_8.194', 'f_8.664', 'f_9.161', 'f_9.687', 'f_10.242', 'f_10.83', 'f_11.451', 'f_12.108', 'f_12.802', 'f_13.536', 'f_14.313', 'f_15.134', 'f_16.002', 'f_16.92', 'f_17.89', 'f_18.916', 'f_20.001', 'f_21.148', 'f_22.361', 'f_23.644', 'f_25.0']\n"
     ]
    }
   ],
   "source": [
    "print(df_ew2.columns.tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "official-respect",
   "metadata": {},
   "outputs": [],
   "source": [
    "df=df_ew2[['MT_Magnitude_','rjb_0','Vs30','Code', 'Address','MT_Depth_','Region_Name_', 'samplingFreq',\n",
    "           'Tectonic_Garcia_', 'Tectonic_Zhoa_', 'JMA_Depth_','Focal_mechanism_kegan_', 'Strike_', 'recordDate',\n",
    "           'Number_of_Stations_','Dip_', 'direction','Focal_mechanism_Garcia_', 'Rake_', 'duration Time',\n",
    "           'D5_75_NS2','D20_80_NS2','D5_95_NS2','D5_75_EW2','D20_80_EW2','D5_95_EW2']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "encouraging-tobago",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n",
      "C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  after removing the cwd from sys.path.\n"
     ]
    }
   ],
   "source": [
    "from scipy import stats\n",
    "df['_5_75']=stats.gmean(df.loc[:,['D5_75_EW2','D5_75_NS2']], axis=1)\n",
    "df['_20_80']=stats.gmean(df.loc[:,['D20_80_EW2','D20_80_NS2']], axis=1)\n",
    "df['_5_95']=stats.gmean(df.loc[:,['D5_95_EW2','D5_95_NS2']], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "arabic-morocco",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(59011, 30)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df=df.reset_index()\n",
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "printable-nicaragua",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "index                      False\n",
       "MT_Magnitude_              False\n",
       "rjb_0                      False\n",
       "Vs30                        True\n",
       "Code                       False\n",
       "Address                    False\n",
       "MT_Depth_                  False\n",
       "Region_Name_               False\n",
       "samplingFreq               False\n",
       "Tectonic_Garcia_           False\n",
       "Tectonic_Zhoa_             False\n",
       "JMA_Depth_                 False\n",
       "Focal_mechanism_kegan_     False\n",
       "Strike_                    False\n",
       "recordDate                 False\n",
       "Number_of_Stations_        False\n",
       "Dip_                       False\n",
       "direction                  False\n",
       "Focal_mechanism_Garcia_    False\n",
       "Rake_                      False\n",
       "duration Time              False\n",
       "D5_75_NS2                  False\n",
       "D20_80_NS2                 False\n",
       "D5_95_NS2                  False\n",
       "D5_75_EW2                  False\n",
       "D20_80_EW2                 False\n",
       "D5_95_EW2                  False\n",
       "_5_75                      False\n",
       "_20_80                     False\n",
       "_5_95                      False\n",
       "dtype: bool"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.isnull().any()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "surprising-longer",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.dropna(axis=0,inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "authentic-findings",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(53797, 30)"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "welsh-development",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-19-01926316919d>\", line 1, in <module>\n",
      "    plt.scatterplot(df['Vs30'],df['MT_Magnitude_'])\n",
      "AttributeError: module 'matplotlib.pyplot' has no attribute 'scatterplot'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'AttributeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-19-01926316919d>\", line 1, in <module>\n",
      "    plt.scatterplot(df['Vs30'],df['MT_Magnitude_'])\n",
      "AttributeError: module 'matplotlib.pyplot' has no attribute 'scatterplot'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'AttributeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3347, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3444, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1125, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-19-01926316919d>\", line 1, in <module>\n",
      "    plt.scatterplot(df['Vs30'],df['MT_Magnitude_'])\n",
      "AttributeError: module 'matplotlib.pyplot' has no attribute 'scatterplot'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'AttributeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3347, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3444, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1125, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2932, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3156, in run_cell_async\n",
      "    interactivity=interactivity, compiler=compiler, result=result)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3366, in run_ast_nodes\n",
      "    self.showtraceback()\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1143, in structured_traceback\n",
      "    chained_exceptions_tb_offset)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "ERROR! Session/line number was not unique in database. History logging moved to new session 936\n"
     ]
    }
   ],
   "source": [
    "plt.scatterplot(df['Vs30'],df['MT_Magnitude_'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "legal-charger",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>MT_Magnitude_</th>\n",
       "      <th>rjb_0</th>\n",
       "      <th>Vs30</th>\n",
       "      <th>Code</th>\n",
       "      <th>Address</th>\n",
       "      <th>MT_Depth_</th>\n",
       "      <th>Region_Name_</th>\n",
       "      <th>samplingFreq</th>\n",
       "      <th>Tectonic_Garcia_</th>\n",
       "      <th>Tectonic_Zhoa_</th>\n",
       "      <th>JMA_Depth_</th>\n",
       "      <th>Focal_mechanism_kegan_</th>\n",
       "      <th>Strike_</th>\n",
       "      <th>recordDate</th>\n",
       "      <th>Number_of_Stations_</th>\n",
       "      <th>Dip_</th>\n",
       "      <th>direction</th>\n",
       "      <th>Focal_mechanism_Garcia_</th>\n",
       "      <th>Rake_</th>\n",
       "      <th>duration Time</th>\n",
       "      <th>D5_75_NS2</th>\n",
       "      <th>D20_80_NS2</th>\n",
       "      <th>D5_95_NS2</th>\n",
       "      <th>D5_75_EW2</th>\n",
       "      <th>D20_80_EW2</th>\n",
       "      <th>D5_95_EW2</th>\n",
       "      <th>_5_75</th>\n",
       "      <th>_20_80</th>\n",
       "      <th>_5_95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>201.366205</td>\n",
       "      <td>241.153846</td>\n",
       "      <td>AICH049807010222</td>\n",
       "      <td>['/19980701022200/AICH049807010222']</td>\n",
       "      <td>8</td>\n",
       "      <td>NORTHERN_NAGANO_PREF</td>\n",
       "      <td>200</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>8.57</td>\n",
       "      <td>1</td>\n",
       "      <td>354;244</td>\n",
       "      <td>1998-01-07 00:00:00</td>\n",
       "      <td>3</td>\n",
       "      <td>71;47</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>46;153</td>\n",
       "      <td>60</td>\n",
       "      <td>18.755</td>\n",
       "      <td>17.205</td>\n",
       "      <td>32.665</td>\n",
       "      <td>15.670</td>\n",
       "      <td>14.035</td>\n",
       "      <td>30.665</td>\n",
       "      <td>17.143245</td>\n",
       "      <td>15.539375</td>\n",
       "      <td>31.649206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>5.0</td>\n",
       "      <td>212.569993</td>\n",
       "      <td>301.324503</td>\n",
       "      <td>AICH059807010222</td>\n",
       "      <td>['/19980701022200/AICH059807010222']</td>\n",
       "      <td>8</td>\n",
       "      <td>NORTHERN_NAGANO_PREF</td>\n",
       "      <td>200</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>8.57</td>\n",
       "      <td>1</td>\n",
       "      <td>354;244</td>\n",
       "      <td>1998-01-07 00:00:00</td>\n",
       "      <td>3</td>\n",
       "      <td>71;47</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>46;153</td>\n",
       "      <td>89</td>\n",
       "      <td>25.350</td>\n",
       "      <td>11.890</td>\n",
       "      <td>39.275</td>\n",
       "      <td>26.860</td>\n",
       "      <td>12.985</td>\n",
       "      <td>41.555</td>\n",
       "      <td>26.094080</td>\n",
       "      <td>12.425444</td>\n",
       "      <td>40.398919</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>5.0</td>\n",
       "      <td>160.594546</td>\n",
       "      <td>428.100722</td>\n",
       "      <td>AICH079807010222</td>\n",
       "      <td>['/19980701022200/AICH079807010222']</td>\n",
       "      <td>8</td>\n",
       "      <td>NORTHERN_NAGANO_PREF</td>\n",
       "      <td>200</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>8.57</td>\n",
       "      <td>1</td>\n",
       "      <td>354;244</td>\n",
       "      <td>1998-01-07 00:00:00</td>\n",
       "      <td>3</td>\n",
       "      <td>71;47</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>46;153</td>\n",
       "      <td>77</td>\n",
       "      <td>19.135</td>\n",
       "      <td>5.295</td>\n",
       "      <td>25.105</td>\n",
       "      <td>4.320</td>\n",
       "      <td>3.665</td>\n",
       "      <td>9.920</td>\n",
       "      <td>9.091930</td>\n",
       "      <td>4.405244</td>\n",
       "      <td>15.781052</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>5.0</td>\n",
       "      <td>195.256405</td>\n",
       "      <td>448.533640</td>\n",
       "      <td>AICH089807010222</td>\n",
       "      <td>['/19980701022200/AICH089807010222']</td>\n",
       "      <td>8</td>\n",
       "      <td>NORTHERN_NAGANO_PREF</td>\n",
       "      <td>200</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>8.57</td>\n",
       "      <td>1</td>\n",
       "      <td>354;244</td>\n",
       "      <td>1998-01-07 00:00:00</td>\n",
       "      <td>3</td>\n",
       "      <td>71;47</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>46;153</td>\n",
       "      <td>77</td>\n",
       "      <td>26.155</td>\n",
       "      <td>6.425</td>\n",
       "      <td>33.085</td>\n",
       "      <td>15.705</td>\n",
       "      <td>4.245</td>\n",
       "      <td>21.410</td>\n",
       "      <td>20.267320</td>\n",
       "      <td>5.222463</td>\n",
       "      <td>26.614843</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>5.0</td>\n",
       "      <td>220.505360</td>\n",
       "      <td>274.027568</td>\n",
       "      <td>AICH099807010222</td>\n",
       "      <td>['/19980701022200/AICH099807010222']</td>\n",
       "      <td>8</td>\n",
       "      <td>NORTHERN_NAGANO_PREF</td>\n",
       "      <td>200</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>8.57</td>\n",
       "      <td>1</td>\n",
       "      <td>354;244</td>\n",
       "      <td>1998-01-07 00:00:00</td>\n",
       "      <td>3</td>\n",
       "      <td>71;47</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>46;153</td>\n",
       "      <td>84</td>\n",
       "      <td>31.625</td>\n",
       "      <td>10.610</td>\n",
       "      <td>43.535</td>\n",
       "      <td>28.405</td>\n",
       "      <td>11.245</td>\n",
       "      <td>42.955</td>\n",
       "      <td>29.971789</td>\n",
       "      <td>10.922887</td>\n",
       "      <td>43.244028</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59006</th>\n",
       "      <td>59006</td>\n",
       "      <td>3.6</td>\n",
       "      <td>48.584792</td>\n",
       "      <td>371.462264</td>\n",
       "      <td>TCGH101209301441</td>\n",
       "      <td>['/20120930144100/TCGH101209301441']</td>\n",
       "      <td>8</td>\n",
       "      <td>NORTHERN_IBARAKI_PREF</td>\n",
       "      <td>100</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>7.85</td>\n",
       "      <td>3</td>\n",
       "      <td>176;2</td>\n",
       "      <td>9/30/2012</td>\n",
       "      <td>3</td>\n",
       "      <td>50;40</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>-94;-85</td>\n",
       "      <td>120</td>\n",
       "      <td>4.490</td>\n",
       "      <td>3.750</td>\n",
       "      <td>13.260</td>\n",
       "      <td>3.460</td>\n",
       "      <td>1.760</td>\n",
       "      <td>9.290</td>\n",
       "      <td>3.941497</td>\n",
       "      <td>2.569047</td>\n",
       "      <td>11.098892</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59007</th>\n",
       "      <td>59007</td>\n",
       "      <td>3.6</td>\n",
       "      <td>70.884798</td>\n",
       "      <td>328.979157</td>\n",
       "      <td>TCGH111209301441</td>\n",
       "      <td>['/20120930144100/TCGH111209301441']</td>\n",
       "      <td>8</td>\n",
       "      <td>NORTHERN_IBARAKI_PREF</td>\n",
       "      <td>100</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>7.85</td>\n",
       "      <td>3</td>\n",
       "      <td>176;2</td>\n",
       "      <td>9/30/2012</td>\n",
       "      <td>3</td>\n",
       "      <td>50;40</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>-94;-85</td>\n",
       "      <td>120</td>\n",
       "      <td>5.600</td>\n",
       "      <td>6.870</td>\n",
       "      <td>17.760</td>\n",
       "      <td>6.520</td>\n",
       "      <td>6.360</td>\n",
       "      <td>19.840</td>\n",
       "      <td>6.042516</td>\n",
       "      <td>6.610083</td>\n",
       "      <td>18.771212</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59008</th>\n",
       "      <td>59008</td>\n",
       "      <td>3.6</td>\n",
       "      <td>52.028046</td>\n",
       "      <td>343.678161</td>\n",
       "      <td>TCGH121209301441</td>\n",
       "      <td>['/20120930144100/TCGH121209301441']</td>\n",
       "      <td>8</td>\n",
       "      <td>NORTHERN_IBARAKI_PREF</td>\n",
       "      <td>100</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>7.85</td>\n",
       "      <td>3</td>\n",
       "      <td>176;2</td>\n",
       "      <td>9/30/2012</td>\n",
       "      <td>3</td>\n",
       "      <td>50;40</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>-94;-85</td>\n",
       "      <td>120</td>\n",
       "      <td>4.810</td>\n",
       "      <td>5.460</td>\n",
       "      <td>15.090</td>\n",
       "      <td>5.640</td>\n",
       "      <td>6.250</td>\n",
       "      <td>18.750</td>\n",
       "      <td>5.208493</td>\n",
       "      <td>5.841661</td>\n",
       "      <td>16.820746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59009</th>\n",
       "      <td>59009</td>\n",
       "      <td>3.6</td>\n",
       "      <td>34.276865</td>\n",
       "      <td>573.596358</td>\n",
       "      <td>TCGH131209301441</td>\n",
       "      <td>['/20120930144100/TCGH131209301441']</td>\n",
       "      <td>8</td>\n",
       "      <td>NORTHERN_IBARAKI_PREF</td>\n",
       "      <td>100</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>7.85</td>\n",
       "      <td>3</td>\n",
       "      <td>176;2</td>\n",
       "      <td>9/30/2012</td>\n",
       "      <td>3</td>\n",
       "      <td>50;40</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>-94;-85</td>\n",
       "      <td>120</td>\n",
       "      <td>2.530</td>\n",
       "      <td>2.600</td>\n",
       "      <td>8.140</td>\n",
       "      <td>3.940</td>\n",
       "      <td>4.190</td>\n",
       "      <td>11.120</td>\n",
       "      <td>3.157246</td>\n",
       "      <td>3.300606</td>\n",
       "      <td>9.514032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59010</th>\n",
       "      <td>59010</td>\n",
       "      <td>3.6</td>\n",
       "      <td>49.819778</td>\n",
       "      <td>213.197970</td>\n",
       "      <td>TCGH161209301441</td>\n",
       "      <td>['/20120930144100/TCGH161209301441']</td>\n",
       "      <td>8</td>\n",
       "      <td>NORTHERN_IBARAKI_PREF</td>\n",
       "      <td>100</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>7.85</td>\n",
       "      <td>3</td>\n",
       "      <td>176;2</td>\n",
       "      <td>9/30/2012</td>\n",
       "      <td>3</td>\n",
       "      <td>50;40</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>-94;-85</td>\n",
       "      <td>120</td>\n",
       "      <td>6.630</td>\n",
       "      <td>6.650</td>\n",
       "      <td>16.980</td>\n",
       "      <td>3.950</td>\n",
       "      <td>4.890</td>\n",
       "      <td>15.170</td>\n",
       "      <td>5.117470</td>\n",
       "      <td>5.702499</td>\n",
       "      <td>16.049505</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>59011 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       index  MT_Magnitude_       rjb_0        Vs30              Code  \\\n",
       "0          0            5.0  201.366205  241.153846  AICH049807010222   \n",
       "1          1            5.0  212.569993  301.324503  AICH059807010222   \n",
       "2          2            5.0  160.594546  428.100722  AICH079807010222   \n",
       "3          3            5.0  195.256405  448.533640  AICH089807010222   \n",
       "4          4            5.0  220.505360  274.027568  AICH099807010222   \n",
       "...      ...            ...         ...         ...               ...   \n",
       "59006  59006            3.6   48.584792  371.462264  TCGH101209301441   \n",
       "59007  59007            3.6   70.884798  328.979157  TCGH111209301441   \n",
       "59008  59008            3.6   52.028046  343.678161  TCGH121209301441   \n",
       "59009  59009            3.6   34.276865  573.596358  TCGH131209301441   \n",
       "59010  59010            3.6   49.819778  213.197970  TCGH161209301441   \n",
       "\n",
       "                                    Address  MT_Depth_           Region_Name_  \\\n",
       "0      ['/19980701022200/AICH049807010222']          8   NORTHERN_NAGANO_PREF   \n",
       "1      ['/19980701022200/AICH059807010222']          8   NORTHERN_NAGANO_PREF   \n",
       "2      ['/19980701022200/AICH079807010222']          8   NORTHERN_NAGANO_PREF   \n",
       "3      ['/19980701022200/AICH089807010222']          8   NORTHERN_NAGANO_PREF   \n",
       "4      ['/19980701022200/AICH099807010222']          8   NORTHERN_NAGANO_PREF   \n",
       "...                                     ...        ...                    ...   \n",
       "59006  ['/20120930144100/TCGH101209301441']          8  NORTHERN_IBARAKI_PREF   \n",
       "59007  ['/20120930144100/TCGH111209301441']          8  NORTHERN_IBARAKI_PREF   \n",
       "59008  ['/20120930144100/TCGH121209301441']          8  NORTHERN_IBARAKI_PREF   \n",
       "59009  ['/20120930144100/TCGH131209301441']          8  NORTHERN_IBARAKI_PREF   \n",
       "59010  ['/20120930144100/TCGH161209301441']          8  NORTHERN_IBARAKI_PREF   \n",
       "\n",
       "       samplingFreq  Tectonic_Garcia_  Tectonic_Zhoa_  JMA_Depth_  \\\n",
       "0               200                 2               2        8.57   \n",
       "1               200                 2               2        8.57   \n",
       "2               200                 2               2        8.57   \n",
       "3               200                 2               2        8.57   \n",
       "4               200                 2               2        8.57   \n",
       "...             ...               ...             ...         ...   \n",
       "59006           100                 2               2        7.85   \n",
       "59007           100                 2               2        7.85   \n",
       "59008           100                 2               2        7.85   \n",
       "59009           100                 2               2        7.85   \n",
       "59010           100                 2               2        7.85   \n",
       "\n",
       "       Focal_mechanism_kegan_  Strike_           recordDate  \\\n",
       "0                           1  354;244  1998-01-07 00:00:00   \n",
       "1                           1  354;244  1998-01-07 00:00:00   \n",
       "2                           1  354;244  1998-01-07 00:00:00   \n",
       "3                           1  354;244  1998-01-07 00:00:00   \n",
       "4                           1  354;244  1998-01-07 00:00:00   \n",
       "...                       ...      ...                  ...   \n",
       "59006                       3    176;2            9/30/2012   \n",
       "59007                       3    176;2            9/30/2012   \n",
       "59008                       3    176;2            9/30/2012   \n",
       "59009                       3    176;2            9/30/2012   \n",
       "59010                       3    176;2            9/30/2012   \n",
       "\n",
       "       Number_of_Stations_   Dip_  direction  Focal_mechanism_Garcia_  \\\n",
       "0                        3  71;47          2                        0   \n",
       "1                        3  71;47          2                        0   \n",
       "2                        3  71;47          2                        0   \n",
       "3                        3  71;47          2                        0   \n",
       "4                        3  71;47          2                        0   \n",
       "...                    ...    ...        ...                      ...   \n",
       "59006                    3  50;40          2                        3   \n",
       "59007                    3  50;40          2                        3   \n",
       "59008                    3  50;40          2                        3   \n",
       "59009                    3  50;40          2                        3   \n",
       "59010                    3  50;40          2                        3   \n",
       "\n",
       "         Rake_  duration Time  D5_75_NS2  D20_80_NS2  D5_95_NS2  D5_75_EW2  \\\n",
       "0       46;153             60     18.755      17.205     32.665     15.670   \n",
       "1       46;153             89     25.350      11.890     39.275     26.860   \n",
       "2       46;153             77     19.135       5.295     25.105      4.320   \n",
       "3       46;153             77     26.155       6.425     33.085     15.705   \n",
       "4       46;153             84     31.625      10.610     43.535     28.405   \n",
       "...        ...            ...        ...         ...        ...        ...   \n",
       "59006  -94;-85            120      4.490       3.750     13.260      3.460   \n",
       "59007  -94;-85            120      5.600       6.870     17.760      6.520   \n",
       "59008  -94;-85            120      4.810       5.460     15.090      5.640   \n",
       "59009  -94;-85            120      2.530       2.600      8.140      3.940   \n",
       "59010  -94;-85            120      6.630       6.650     16.980      3.950   \n",
       "\n",
       "       D20_80_EW2  D5_95_EW2      _5_75     _20_80      _5_95  \n",
       "0          14.035     30.665  17.143245  15.539375  31.649206  \n",
       "1          12.985     41.555  26.094080  12.425444  40.398919  \n",
       "2           3.665      9.920   9.091930   4.405244  15.781052  \n",
       "3           4.245     21.410  20.267320   5.222463  26.614843  \n",
       "4          11.245     42.955  29.971789  10.922887  43.244028  \n",
       "...           ...        ...        ...        ...        ...  \n",
       "59006       1.760      9.290   3.941497   2.569047  11.098892  \n",
       "59007       6.360     19.840   6.042516   6.610083  18.771212  \n",
       "59008       6.250     18.750   5.208493   5.841661  16.820746  \n",
       "59009       4.190     11.120   3.157246   3.300606   9.514032  \n",
       "59010       4.890     15.170   5.117470   5.702499  16.049505  \n",
       "\n",
       "[59011 rows x 30 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "sensitive-sheet",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "299.968812"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['rjb_0'].max()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "significant-omega",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MT_Magnitude_</th>\n",
       "      <th>rjb_0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5.0</td>\n",
       "      <td>201.366205</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5.0</td>\n",
       "      <td>212.569993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5.0</td>\n",
       "      <td>160.594546</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5.0</td>\n",
       "      <td>195.256405</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5.0</td>\n",
       "      <td>220.505360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59006</th>\n",
       "      <td>3.6</td>\n",
       "      <td>48.584792</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59007</th>\n",
       "      <td>3.6</td>\n",
       "      <td>70.884798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59008</th>\n",
       "      <td>3.6</td>\n",
       "      <td>52.028046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59009</th>\n",
       "      <td>3.6</td>\n",
       "      <td>34.276865</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59010</th>\n",
       "      <td>3.6</td>\n",
       "      <td>49.819778</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>59011 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       MT_Magnitude_       rjb_0\n",
       "0                5.0  201.366205\n",
       "1                5.0  212.569993\n",
       "2                5.0  160.594546\n",
       "3                5.0  195.256405\n",
       "4                5.0  220.505360\n",
       "...              ...         ...\n",
       "59006            3.6   48.584792\n",
       "59007            3.6   70.884798\n",
       "59008            3.6   52.028046\n",
       "59009            3.6   34.276865\n",
       "59010            3.6   49.819778\n",
       "\n",
       "[59011 rows x 2 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "input_var=df[['MT_Magnitude_','rjb_0']]\n",
    "input_var"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "heavy-barrier",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "MT_Magnitude_    0\n",
       "rjb_0            0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ERROR! Session/line number was not unique in database. History logging moved to new session 930\n"
     ]
    }
   ],
   "source": [
    "input_var.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "vocational-gardening",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-14-ce479c084262>\", line 1, in <module>\n",
      "    sns.heatmap(input_var.isnull(), ytickslabels=False, cbar=False, cmap='viridis')\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\seaborn\\matrix.py\", line 548, in heatmap\n",
      "    plotter.plot(ax, cbar_ax, kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\seaborn\\matrix.py\", line 302, in plot\n",
      "    cmap=self.cmap, **kws)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\__init__.py\", line 1565, in inner\n",
      "    return func(ax, *map(sanitize_sequence, args), **kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\axes\\_axes.py\", line 6118, in pcolormesh\n",
      "    **kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\collections.py\", line 1941, in __init__\n",
      "    Collection.__init__(self, **kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\collections.py\", line 164, in __init__\n",
      "    self.update(kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\artist.py\", line 1006, in update\n",
      "    ret = [_update_property(self, k, v) for k, v in props.items()]\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\artist.py\", line 1006, in <listcomp>\n",
      "    ret = [_update_property(self, k, v) for k, v in props.items()]\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\artist.py\", line 1002, in _update_property\n",
      "    .format(type(self).__name__, k))\n",
      "AttributeError: 'QuadMesh' object has no property 'ytickslabels'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'AttributeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-14-ce479c084262>\", line 1, in <module>\n",
      "    sns.heatmap(input_var.isnull(), ytickslabels=False, cbar=False, cmap='viridis')\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\seaborn\\matrix.py\", line 548, in heatmap\n",
      "    plotter.plot(ax, cbar_ax, kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\seaborn\\matrix.py\", line 302, in plot\n",
      "    cmap=self.cmap, **kws)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\__init__.py\", line 1565, in inner\n",
      "    return func(ax, *map(sanitize_sequence, args), **kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\axes\\_axes.py\", line 6118, in pcolormesh\n",
      "    **kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\collections.py\", line 1941, in __init__\n",
      "    Collection.__init__(self, **kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\collections.py\", line 164, in __init__\n",
      "    self.update(kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\artist.py\", line 1006, in update\n",
      "    ret = [_update_property(self, k, v) for k, v in props.items()]\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\artist.py\", line 1006, in <listcomp>\n",
      "    ret = [_update_property(self, k, v) for k, v in props.items()]\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\artist.py\", line 1002, in _update_property\n",
      "    .format(type(self).__name__, k))\n",
      "AttributeError: 'QuadMesh' object has no property 'ytickslabels'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'AttributeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3347, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3444, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1125, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-14-ce479c084262>\", line 1, in <module>\n",
      "    sns.heatmap(input_var.isnull(), ytickslabels=False, cbar=False, cmap='viridis')\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\seaborn\\matrix.py\", line 548, in heatmap\n",
      "    plotter.plot(ax, cbar_ax, kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\seaborn\\matrix.py\", line 302, in plot\n",
      "    cmap=self.cmap, **kws)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\__init__.py\", line 1565, in inner\n",
      "    return func(ax, *map(sanitize_sequence, args), **kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\axes\\_axes.py\", line 6118, in pcolormesh\n",
      "    **kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\collections.py\", line 1941, in __init__\n",
      "    Collection.__init__(self, **kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\collections.py\", line 164, in __init__\n",
      "    self.update(kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\artist.py\", line 1006, in update\n",
      "    ret = [_update_property(self, k, v) for k, v in props.items()]\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\artist.py\", line 1006, in <listcomp>\n",
      "    ret = [_update_property(self, k, v) for k, v in props.items()]\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\matplotlib\\artist.py\", line 1002, in _update_property\n",
      "    .format(type(self).__name__, k))\n",
      "AttributeError: 'QuadMesh' object has no property 'ytickslabels'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'AttributeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3347, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3444, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1125, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2932, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3156, in run_cell_async\n",
      "    interactivity=interactivity, compiler=compiler, result=result)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3366, in run_ast_nodes\n",
      "    self.showtraceback()\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1143, in structured_traceback\n",
      "    chained_exceptions_tb_offset)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\Dev\\anaconda3\\envs\\tf\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXwAAAD8CAYAAAB0IB+mAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAANAElEQVR4nO3cX4idd53H8fdnEwOK0harriQpdiX+6YWFWtsiulRLMelNEHrRKhaKkC1LxcsWL3TBm/VOxD9hKKHIgrnQonFRi7BohZrdRGjTpiVhjNDORii1RaFelLTfvZjTzThO5jxz5pwz6XzfL3ggz3l+55zf/Dp595knz5xUFZKk7e8ftnoCkqT5MPiS1ITBl6QmDL4kNWHwJakJgy9JTYwNfpIjSV5I8vQljifJt5IsJjmV5IbpT1OStFlDzvAfBvavc/wAsG+0HQK+t/lpSZKmbWzwq+ox4KV1hhwEvl/LjgNXJnnvtCYoSZqOaVzD3w08v2J/afTY30lyKMnJJCcXFhYKcHNzc3Pb2DaxnZt58kjWeGzNSVXVArCw3hhJ0mxM4wx/Cdi7Yn8PcH4KrytJmqJpBP8YcM/obp1bgD9X1R+n8LqSpCkae0knyQ+AW4GrkywBXwPeAlBVh4GfAXcAi8BfgXtnNVlJ0uSyhR+P7DV8Sdq4tf7ddBB/01aSmjD4ktSEwZekJgy+JDVh8CWpCYMvSU0YfElqwuBLUhMGX5KaMPiS1ITBl6QmDL4kNWHwJakJgy9JTRh8SWrC4EtSEwZfkpow+JLUhMGXpCYMviQ1YfAlqQmDL0lNGHxJasLgS1ITBl+SmjD4ktSEwZekJgy+JDVh8CWpCYMvSU0YfElqwuBLUhMGX5KaMPiS1ITBl6QmBgU/yf4kZ5IsJnlwjeNXJPlpkieTnE5y7/SnKknajFTV+gOSHcBZ4HZgCTgB3F1Vz6wY8xXgiqp6IMm7gDPAP1bVq+u89PpvLElaSyZ94pAz/JuAxao6Nwr4UeDgqjEFvCNJgLcDLwEXJp2UJGn6hgR/N/D8iv2l0WMrfRv4MHAeeAr4clW9vvqFkhxKcjLJyYWFhQmnLEmaxM4BY9b68WH15ZjPAE8AnwbeD/wyyW+q6i9/86SqBeCN0ntJR5LmaMgZ/hKwd8X+HpbP5Fe6F3ikli0CfwA+NJ0pSpKmYUjwTwD7klybZBdwF3Bs1ZjngNsAkrwH+CBwbpoTlSRtzthLOlV1Icn9wKPADuBIVZ1Oct/o+GHg68DDSZ5i+RLQA1X14gznLUnaoLG3Zc6Q1/AlaeNmelumJGkbMPiS1ITBl6QmDL4kNWHwJakJgy9JTRh8SWrC4EtSEwZfkpow+JLUhMGXpCYMviQ1YfAlqQmDL0lNGHxJasLgS1ITBl+SmjD4ktSEwZekJgy+JDVh8CWpCYMvSU0YfElqwuBLUhMGX5KaMPiS1ITBl6QmDL4kNWHwJakJgy9JTRh8SWrC4EtSEwZfkpow+JLUxKDgJ9mf5EySxSQPXmLMrUmeSHI6ya+nO01J0malqtYfkOwAzgK3A0vACeDuqnpmxZgrgceB/VX1XJJ3V9ULY957/TeWJK0lkz5xyBn+TcBiVZ2rqleBo8DBVWM+BzxSVc8BDIi9JGnOhgR/N/D8iv2l0WMrfQC4KsmvkvwuyT1rvVCSQ0lOJjm5sLAw2YwlSRPZOWDMWj8+rL4csxP4KHAb8Fbgt0mOV9XZv3lS1QLwRum9pCNJczQk+EvA3hX7e4Dza4x5sapeAV5J8hhwPcvX/iVJl4Ehl3ROAPuSXJtkF3AXcGzVmJ8An0yyM8nbgJuBZ6c7VUnSZow9w6+qC0nuBx4FdgBHqup0kvtGxw9X1bNJfgGcAl4HHqqqp2c5cUnSxoy9LXOGvIYvSRs309syJUnbgMGXpCYMviQ1YfAlqQmDL0lNGHxJasLgS1ITBl+SmjD4ktSEwZekJgy+JDVh8CWpCYMvSU0YfElqwuBLUhMGX5KaMPiS1ITBl6QmDL4kNWHwJakJgy9JTRh8SWrC4EtSEwZfkpow+JLUhMGXpCYMviQ1YfAlqQmDL0lNGHxJasLgS1ITBl+SmjD4ktSEwZekJgy+JDUxKPhJ9ic5k2QxyYPrjPtYkteS3Dm9KUqSpmFs8JPsAL4DHACuA+5Oct0lxn0DeHTak5Qkbd6QM/ybgMWqOldVrwJHgYNrjPsS8CPghSnOT5I0JUOCvxt4fsX+0uix/5dkN/BZ4PB6L5TkUJKTSU4uLCxsdK6SpE3YOWBM1nisVu1/E3igql5L1ho+elLVAvBG6Ve/hiRphoYEfwnYu2J/D3B+1ZgbgaOj2F8N3JHkQlX9eCqzlCRtWqrWP9FOshM4C9wG/C9wAvhcVZ2+xPiHgf+sqh+OeW/P8CVp4y59GWWMsWf4VXUhyf0s332zAzhSVaeT3Dc6vu51e0nS5WHsGf4MeYYvSRs38Rm+v2krSU0YfElqwuBLUhMGX5KaMPiS1ITBl6QmDL4kNWHwJakJgy9JTRh8SWrC4EtSEwZfkpow+JLUhMGXpCYMviQ1YfAlqQmDL0lNGHxJasLgS1ITBl+SmjD4ktSEwZekJgy+JDVh8CWpCYMvSU0YfElqwuBLUhMGX5KaMPiS1ITBl6QmDL4kNWHwJakJgy9JTRh8SWpiUPCT7E9yJslikgfXOP75JKdG2+NJrp/+VCVJm5GqWn9AsgM4C9wOLAEngLur6pkVYz4OPFtVLyc5APxbVd085r3Xf2NJ0loy6ROHnOHfBCxW1bmqehU4ChxcOaCqHq+ql0e7x4E9k05IkjQbQ4K/G3h+xf7S6LFL+SLw87UOJDmU5GSSkwsLC8NnKUnatJ0Dxqz148Oal2OSfIrl4H9ireNVtQC8UXov6UjSHA0J/hKwd8X+HuD86kFJPgI8BByoqj9NZ3qSpGkZcknnBLAvybVJdgF3AcdWDkhyDfAI8IWqOjv9aUqSNmvsGX5VXUhyP/AosAM4UlWnk9w3On4Y+CrwTuC7SQAuVNWNs5u2JGmjxt6WOUNew5ekjZvpbZmSpG3A4EtSEwZfkpow+JLUhMGXpCYMviQ1YfAlqQmDL0lNGHxJasLgS1ITBl+SmjD4ktSEwZekJgy+JDVh8CWpCYMvSU0YfElqwuBLUhMGX5KaMPiS1ITBl6QmDL4kNWHwJakJgy9JTRh8SWrC4EtSEwZfkpow+JLUhMGXpCYMviQ1YfAlqQmDL0lNGHxJasLgS1ITBl+SmhgU/CT7k5xJspjkwTWOJ8m3RsdPJblh+lOVJG3G2OAn2QF8BzgAXAfcneS6VcMOAPtG2yHge1OepyRpk4ac4d8ELFbVuap6FTgKHFw15iDw/Vp2HLgyyXunPFdJ0iYMCf5u4PkV+0ujxzY6hiSHkpwcbf8BxI0k+ZetnsPlsrkWroVrMXYtDjGhIcHPGo/VBGOoqoWqurGqbgQ+POC9u5j4P+A25Fpc5Fpc5FpcNNPgLwF7V+zvAc5PMEaStIWGBP8EsC/JtUl2AXcBx1aNOQbcM7pb5xbgz1X1xynPVZK0CTvHDaiqC0nuBx4FdgBHqup0kvtGxw8DPwPuABaBvwL3DnjvhYlnvf24Fhe5Fhe5Fhe5FhdNvBap+rtL7ZKkbcjftJWkJgy+JDUx8+D7sQwXDViLz4/W4FSSx5NcvxXznIdxa7Fi3MeSvJbkznnOb56GrEWSW5M8keR0kl/Pe47zMuDvyBVJfprkydFaDPn3wjedJEeSvJDk6Uscn6ybVTWzjeV/5P098E/ALuBJ4LpVY+4Afs7yvfy3AP89yzlt1TZwLT4OXDX684HOa7Fi3H+xfFPAnVs97y38vrgSeAa4ZrT/7q2e9xauxVeAb4z+/C7gJWDXVs99Bmvxz8ANwNOXOD5RN2d9hu/HMlw0di2q6vGqenm0e5zl32fYjoZ8XwB8CfgR8MI8JzdnQ9bic8AjVfUcQFVt1/UYshYFvCNJgLezHPwL853m7FXVYyx/bZcyUTdnHfypfSzDNrDRr/OLLP8ffDsauxZJdgOfBQ7PcV5bYcj3xQeAq5L8Ksnvktwzt9nN15C1+DbLv6V/HngK+HJVvT6f6V1WJurm2PvwN2lqH8uwDQz+OpN8iuXgf2KmM9o6Q9bim8ADVfXa8snctjVkLXYCHwVuA94K/DbJ8ao6O+vJzdmQtfgM8ATwaeD9wC+T/Kaq/jLryV1mJurmrIPvxzJcNOjrTPIR4CHgQFX9aU5zm7cha3EjcHQU+6uBO5JcqKofz2eKczP078iLVfUK8EqSx4Drge0W/CFrcS/w77V8IXsxyR+ADwH/M58pXjYm6uasL+n4sQwXjV2LJNcAjwBf2IZnbyuNXYuquraq3ldV7wN+CPzrNow9DPs78hPgk0l2JnkbcDPw7JznOQ9D1uI5ln/SIcl7gA8C5+Y6y8vDRN2c6Rl+ze5jGd50Bq7FV4F3At8dndleqOVPFt1WBq5FC0PWoqqeTfIL4BTwOvBQVa15u96b2cDvi68DDyd5iuXLGg9U1YtbNukZSfID4Fbg6iRLwNeAt8DmuulHK0hSE/6mrSQ1YfAlqQmDL0lNGHxJasLgS1ITBl+SmjD4ktTE/wEvjqMNqRcmawAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "sns.heatmap(input_var.isnull(), ytickslabels=False, cbar=False, cmap='viridis')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "following-candy",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
